### PR TITLE
[feat] - feat(front): workspace analytics UI

### LINKED ISSUE #5966: [Workspace Metrics] - UI

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR introduces improvements to the workspace analytics UI: https://github.com/dust-tt/tasks/issues/897

1. Implementing a new data download system using `JSZip` to generate and download multiple CSV files containing workspace usage data.
2. Updating the UI for the Full Activity Report section, allowing users to download data for specific months.

**New look:**
Screenshot 2024-07-01 at 17 33 46

## Risk
1. The introduction of JSZip for file compression and download may impact performance for large datasets.
2. New queries and data processing logic could potentially increase the load on our databases.

## Deploy Plan
1. Deploy to `front-edge` and check stress load on DB.
2. Deploy to `front`

### CODE REVIEW

**Comment by JulesBelveze** [comment]
@Duncid I took the freedom to slightly change a couple of things:
- I removed the "All" button from your design: I am afraid that this is gonna be too time-consuming and gonna stress the DB too much
- I added a "Download" button: the download can take 5-10sec, so to make sure the user doesn't click on it multiple I disable buttons while the requests are running.

Are you okay with it?

**Comment by thib-martin** [comment]
Thanks @JulesBelveze, a few comments:
* I assume front-edge is on fake data? I see months in the future which I assume you would not get?
* Possible to sort by most recent month (if not planned already)?
* CSV breaks for now, because some descriptions have comas in them. Maybe let's find another delimiter which is more bullet-proof?
* Ok with not being able to download 'all', However, is it possible to add an idea of which 'month' is being retrieved? Option 1: as a column (easier to digest), option 2: in the file name
* I would rename mentions into assistant_messages (which is a little easier to understand)

**Comment by JulesBelveze** [comment]
@thib-martin let me redeploy with the current state, this will solve a bunch of the comments you made

**Comment by Duncid** [comment]
> Are you okay with it?

Ok for me if @thib-martin is ok with it

**Comment by Duncid** [comment]
Otherwise, LGTM!

File: front/lib/workspace_usage.ts [hunk 1]
```diff
@@ -432,13 +427,29 @@ export async function getAgentUsageData(
...
     | MessageUsageQueryResult[]
     | builderUsageQueryResult[]
 ) {
   const csvHeader = Object.keys(rows[0]).join(",") + "\n";
-  const csvContent = rows.map((row) => Object.values(row).join(",")).join("\n");
+  const csvContent = rows
+    .map((row) =>
+      Object.values(row)
+        .map((value) => {
+          // Escape quotes and wrap in quotes if the value contains commas or quotes
```
Comment (fontanierh) [inline front/lib/workspace_usage.ts:441]: Escaping quotes ourselves is where I draw the limit :D

Let's use this: https://github.com/dust-tt/dust/blob/30404fa506baddafd56df3bd36f1c16caf63697f/front/components/actions/tables_query/TablesQueryActionDetails.tsx#L3

File: front/pages/api/w/[wId]/workspace-usage.ts [hunk 2]
```diff
@@ -15,21 +21,42 @@ const MonthSchema = t.refinement(
...
+export const usageTables = [
+  "users",
+  "assistant_messages",
+  "builders",
+  "agent_configurations",
+  "all",
+];
+type usageTableType = (typeof usageTables)[number];
+
+export function getSupportedUsageTablesCodec(): t.Mixed {
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/workspace-usage.ts:33]: Do you mind adding a comment explaining what this does ? I'm actually quite surprised this is working at all, are you getting any type inference on the decoded values ?

**Review by fontanierh — COMMENTED** [review]
This looks great ! Just a few comments

File: front/pages/api/w/[wId]/workspace-usage.ts (hunk 2 above)
Comment (JulesBelveze) [reply front/pages/api/w/[wId]/workspace-usage.ts:33]: @fontanierh Unfortunately no I am not getting any type here, `query.table` is of type `any`. 
We are doing this in a bunch of places to dynamically create the union type:
- https://github.com/dust-tt/dust/blob/ed7bc31e9e8f3c19ddc4090011454c30368bb11a/front/lib/api/pagination.ts#L24
- https://github.com/dust-tt/dust/blob/ed7bc31e9e8f3c19ddc4090011454c30368bb11a/types/src/front/content_fragment.ts#L52

Would you recommend doing it differently?

Comment (fontanierh) [reply front/pages/api/w/[wId]/workspace-usage.ts:33]: I don't know if there's a better way TBH. Don't want to block for this if we're doing it elsewhere

**Review by fontanierh — APPROVED** [review]

**Review by Duncid — APPROVED** [review]

**Pull request merged**